content/rs/references/memtier-benchmark.md

```diff
@@ -0,0 +1,195 @@
+
+## Data Population
+
+### Populate the benchmark dataset
+
+The memtier_benchmark load generation tool populates the database.
+To populate the database with N items of 500 Bytes each in size, on the load generation instance run:
+
+```src
+$ memtier_benchmark -s $DB_HOST -p $DB_PORT --hide-histogram
+--key-maximum=$N -n allkeys -d 500 --key-pattern=P:P --ratio=1:0
+```
+
+Setup a test database with these values:
+
+|  **Parameter** | **Description** |
+|  ------ | ------ |
+|  Database host<br/>(-s) | The fully qualified name of the endpoint or the IP shown in the RS database configuration |
+|  Database port<br/>(-p) | The endpoint port shown in you RS database configuration |
+|  Number of items<br/>(–key-maximum) | With replication: 75 Million<br/>Without replication: 150 Million |
+|  Item size<br/>(-d) | 500 Bytes |
+
+## Centralize the Keyspace
+
+### With replication
+
+To create about 20.5 million items in RAM for your highly available clustered database with 75 million items, run:
+
+```src
+$ memtier_benchmark  -s $DB_HOST -p $DB_PORT --hide-histogram
+--key-minimum=27250000 --key-maximum=47750000 -n allkeys
+--key-pattern=P:P --ratio=0:1
+```
+
+You can see the **Values in RAM** metric on the **metrics** page of your database in the RS web UI to validate the test.
+
+### Without replication
+
+To create 41 million items in RAM without replication enabled and 150 million items, run:
+
+```src
+$ memtier_benchmark  -s $DB_HOST -p $DB_PORT --hide-histogram
+--key-minimum=54500000 --key-maximum=95500000 -n allkeys
+--key-pattern=P:P --ratio=0:1
+```
+
+## Test Runs
+
+### Generate load
+
+#### With replication
+
+We recommend that you do a dry run and double check the RAM Hit Ratio on the **metrics** screen in the RS web UI before you write down the test results.
+
+To test RoF with an 85% RAM Hit Ratio, run:
+
+```src
+$ memtier_benchmark -s $DB_HOST -p $DB_PORT --pipeline=11 -c 20 -t 1
+-d 500 --key-maximum=75000000 --key-pattern=G:G --key-stddev=5125000
+--ratio=1:1 --distinct-client-seed --randomize --test-time=600
+--run-count=1 --out-file=test.out
+```
+
+#### Without replication
+
+Here is the command for 150 million items:
+
+```src
+$ memtier_benchmark -s $DB_HOST -p $DB_PORT --pipeline=24 -c 20 -t 1
+-d 500 --key-maximum=150000000 --key-pattern=G:G --key-stddev=10250000
+--ratio=1:1 --distinct-client-seed --randomize --test-time=600
+--run-count=1 --out-file=test.out
+```
+
+Where:
+
+|  **Parameter** | **Description** |
+|------------|-----------------|
+|  Access pattern (--key-pattern) and standard deviation (--key-stddev) | Controls the RAM Hit ratio after the centralization process is complete |
+|  Number of threads\ | Controls how many connections are opened to the database, whereby the number of connections is the number of threads multiplied by the number of connections per thread (-t) and number of clients per thread (-c) |
```

Comment:
@Paz-Yanover Do you know why there are 3 entries for Number of Threads?

Reply:
@bmansheim - no. Please consult yaacov.hazan@redislabs.com

Reply:
@yaacov.hazan@redislabs.com Can you explain?

Reply:
Rows are mislabeled.